controllers/gitrepository_controller.go

```diff
@@ -259,6 +261,45 @@ func (r *GitRepositoryReconciler) sync(ctx context.Context, repository sourcev1.
 		return sourcev1.GitRepositoryNotReady(repository, sourcev1.GitOperationFailedReason, err.Error()), err
 	}
 
+	// verify PGP signature
+	if repository.Spec.Verification != nil {
+		commit, err := repo.CommitObject(ref.Hash())
+		if err != nil {
+			err = fmt.Errorf("git resolve HEAD error: %w", err)
+			return sourcev1.GitRepositoryNotReady(repository, sourcev1.GitOperationFailedReason, err.Error()), err
+		}
+
+		if commit.PGPSignature == "" {
+			err = fmt.Errorf("PGP signature not found for commit '%s'", ref.Hash())
+			return sourcev1.GitRepositoryNotReady(repository, sourcev1.GitOperationFailedReason, err.Error()), err
+		}
+
+		name := types.NamespacedName{
+			Namespace: repository.GetNamespace(),
+			Name:      repository.Spec.Verification.SecretRef.Name,
+		}
+
+		var secret corev1.Secret
+		err = r.Client.Get(ctx, name, &secret)
+		if err != nil {
+			err = fmt.Errorf("GPG public keys secret error: %w", err)
```

Comment:
```suggestion
			err = fmt.Errorf("PGP public keys secret error: %w", err)
```

to match with the missing `PGPSignature` error from above.